session-replay/Sources/DatadogSessionReplay/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UITextViewRecorder.swift

```diff
@@ -0,0 +1,84 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import UIKit
+
+internal struct UITextViewRecorder: NodeRecorder {
+    func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeSnapshotBuilder.Context) -> NodeSemantics? {
+        guard let textView = view as? UITextView else {
+            return nil
+        }
+        guard attributes.isVisible else {
+            return InvisibleElement.constant
+        }
+        let builder = UITextViewWireframesBuilder(
+            wireframeID: context.ids.nodeID(for: textView),
+            attributes: attributes,
+            text: textView.text,
+            textColor: textView.textColor?.cgColor ?? UIColor.black.cgColor,
+            font: textView.font,
+            textObfuscator: context.recorder.privacy == .maskAll ? context.textObfuscator : nopTextObfuscator,
+            wireframeRect: CGRect(origin: textView.contentOffset, size: textView.contentSize)
+        )
+        return SpecificElement(wireframesBuilder: builder, recordSubtree: false)
```

Comment:
💡 Interestingly, if the `recordSubtree` is set `true`, it is also able of recording text selection within a text view (just tested it). It behaves weird, as we don't have 1:1 matching font size in the player, but "it works" 😉. I have no opinion on leaving `false` and ignoring text selection vs setting `true` and having it done in an imperfect way for now.

Reply:
Yeah - good point. I'll add it. It's very hard to fully support when take scaling into account, but usually TextView is used with scrollview enabled, so it should be fine for most cases when we update the font.